tests/bluetooth/bsim_bt/bsim_test_mesh/prj.conf

```diff
@@ -38,6 +38,6 @@ CONFIG_BT_MESH_PB_ADV=y
 CONFIG_BT_MESH_PROVISIONER=y
 CONFIG_BT_MESH_PROV_DEVICE=y
 CONFIG_BT_MESH_CDB=y
-CONFIG_BT_MESH_CDB_NODE_COUNT=4
+CONFIG_BT_MESH_CDB_NODE_COUNT=11
```

Comment:
IMO, it's better to delete the provisioned nodes from the CDB when resetting them.